azurestorage_blueprints/delete_file.py

```diff
@@ -0,0 +1,143 @@
+import os
+import sys
+import re
+import sys
+import argparse
+import shipyard_utils as shipyard
+from azure.storage.blob import BlobClient, ContainerClient
+from azure.core import exceptions
+try:
+    import exit_codes as ec
+except BaseException:
+    from . import exit_codes as ec
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--container-name',
+        dest='container_name',
+        required=True)
+    parser.add_argument(
+        '--source-file-name-match-type',
+        dest='source_file_name_match_type',
+        default='exact_match',
+        choices={
+            'exact_match',
+            'regex_match'},
+        required=False)
+    parser.add_argument(
+        '--source-folder-name',
+        dest='source_folder_name',
+        default='',
+        required=False)
+    parser.add_argument(
+        '--source-file-name',
+        dest='source_file_name',
+        required=True)
+    parser.add_argument(
+        '--connection-string',
+        dest='connection_string',
+        default=None,
+        required=True)
+    return parser.parse_args()
+
+
+def set_environment_variables(args):
+    """
+    Set Azure Connection String as environment variable if it's provided via keyword arguments
+    rather than seeded as environment variables. This will override system defaults.
+    """
+
+    if args.connection_string:
+        os.environ['AZURE_STORAGE_CONNECTION_STRING'] = args.connection_string
+    return
+
+
+def find_azure_storage_blob_file_names(conn_str, container_name, prefix=''):
+    """
+    Fetched all the files in the bucket which are returned in a list as
+    Azure Blob objects
+    """
+    container = ContainerClient.from_connection_string(
+        conn_str=conn_str, container_name=container_name)
+    return list(container.list_blobs(prefix=prefix))
+
+
+def azure_find_matching_files(file_blobs, file_name_re):
+    """
+    Return a list of all file_names that matched the regular expression.
+    """
+    matching_file_names = []
+    for blob in file_blobs:
+        if re.search(file_name_re, blob.name):
+            matching_file_names.append(blob.name)
+
+    return matching_file_names
```

Comment:
Same comment as GCS. Use shipyard_utils files.find_all_file_matches() for consistency across Blueprints

Reply:
This is done now